app/controlplane/pkg/biz/group.go

```diff
@@ -0,0 +1,266 @@
+	SoftDelete(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID) error
+	// ListMembers retrieves a list of members in a group, optionally filtered by maintainer status.
+	ListMembers(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID, opts *ListMembersOpts, paginationOpts *pagination.OffsetPaginationOpts) ([]*GroupMembership, int, error)
+}
+
+// GroupMembership represents a membership of a user in a group.
+type GroupMembership struct {
+	// User is the user who is a member of the group.
+	User *User
+	// Maintainer indicates if the user is a maintainer of the group.
+	Maintainer bool
+	// CreatedAt is the timestamp when the user was added to the group.
+	CreatedAt *time.Time
+	// UpdatedAt is the timestamp when the membership was last updated.
+	UpdatedAt *time.Time
+	// DeletedAt is the timestamp when the membership was deleted, if applicable.
+	DeletedAt *time.Time
+}
+
+type Group struct {
+	// ID is the unique identifier for the group.
+	ID uuid.UUID
+	// Name is the name of the group.
+	Name string
+	// The Description is a brief description of the group.
+	Description string
+	// Members is a list of group memberships, which includes the users who are members of the group.
+	Members []*GroupMembership
+	// Organization is the organization to which the group belongs.
+	Organization *Organization
+	// CreatedAt is the timestamp when the group was created.
+	CreatedAt *time.Time
+	// UpdatedAt is the timestamp when the group was last updated.
+	UpdatedAt *time.Time
+	// DeletedAt is the timestamp when the group was deleted, if applicable.
+	DeletedAt *time.Time
+}
+
+type CreateGroupOpts struct {
+	// Name is the name of the group.
+	Name string
+	// The description is a brief description of the group.
+	Description *string
+	// UserID is the ID of the user who owns the group.
+	UserID uuid.UUID
+}
+
+type UpdateGroupOpts struct {
+	// Description is the new description of the group.
+	Description *string
+}
+
+type ListGroupOpts struct {
+	// Name is the name of the group to filter by.
+	Name string
+	// Description is the description of the group to filter by.
+	Description string
+	// MemberEmail is the email of the member to filter by.
+	MemberEmail string
+}
+
+// ListMembersOpts defines options for listing members of a group.
+type ListMembersOpts struct {
+	// Maintainers indicate whether to filter the members by their maintainer status.
+	Maintainers *bool
+	// MemberEmail is the email of the member to filter by.
+	MemberEmail *string
+}
+
+type GroupUseCase struct {
+	// logger is used to log messages.
+	logger *log.Helper
+	// Repositories
+	groupRepo      GroupRepo
+	membershipRepo MembershipRepo
+	// Auditor use case for logging events
+	auditorUC *AuditorUseCase
+}
+
+func NewGroupUseCase(logger log.Logger, groupRepo GroupRepo, membershipRepo MembershipRepo, auditorUC *AuditorUseCase) *GroupUseCase {
+	return &GroupUseCase{
+		logger:         log.NewHelper(logger),
```

Comment:
in some other places we set the log helper with a namespace no?

Reply:
I will check it and if I find it I'll add it as well here.